frame/staking/src/tests.rs

```diff
@@ -1350,12 +1350,14 @@ fn bond_extra_and_withdraw_unbonded_works() {
 }
 
 #[test]
-fn too_many_unbond_calls_should_not_work() {
+fn many_unbond_calls_should_work() {
```

Comment:
not sure how this test is currently working, but I suggest having new test cases for all possible variants: 

1. unbond, and you have free chunks left.
2. unbond, and you don't have free chunks left.
3. unbond, and you don't have free chunks left, but since some are expired, you still go though.

Reply:
I think all those cases are covered in the current test.

- Case `1. unbond, and you have free chunks left.` and `2.unbond, and you don't have free chunks left.`

Controller account 10 requests one chunk unlocked per era until all the chunks are filled: 
```rust
for i in 0..max_unlocking_chunks - 1 {
	// There is only 1 chunk per era, so we need to be in a new era to create a chunk.
	current_era = i as u32;
	mock::start_active_era(current_era);
	assert_ok!(Staking::unbond(RuntimeOrigin::signed(10), 1));
}

// L1372 -> L1376 checks that the unlocking slots are filled up, as expected
```

- Case `3. unbond, and you don't have free chunks left, but since some are expired, you still go though.`
```rust
// even though the number of unlocked chunks is the same as `MaxUnlockingChunks`,
// unbonding works as expected.
for i in current_era..(current_era + max_unlocking_chunks) - 1 {
	// There is only 1 chunk per era, so we need to be in a new era to create a chunk.
	current_era = i as u32;
	mock::start_active_era(current_era);
	assert_ok!(Staking::unbond(RuntimeOrigin::signed(10), 1));
}

// L1388 -> 1391 checks that the number of free unlocking slots is correct (ie. 3, since unbonding duration is 3). 
```